packages/dds/task-manager/src/test/taskManager.spec.ts

```diff
@@ -425,9 +460,231 @@ describe("TaskManager", () => {
+                    assert.ok(taskManager1.assigned(taskId), "Should be assigned");
+                    taskManager1.abandon(taskId);
+                    assert.ok(!taskManager1.queued(taskId), "Should not be queued");
+                    assert.ok(!taskManager1.assigned(taskId), "Should not be assigned");
+                });
+            });
+
+            describe("Subscribing to a task", () => {
+                it("Can subscribe to a task before attach", async () => {
+                    const taskId = "taskId";
+                    taskManager1.subscribeToTask(taskId);
+                    containerRuntimeFactory.processAllMessages();
+                    assert.ok(taskManager1.queued(taskId), "Task manager 1 should be queued");
+                    assert.ok(taskManager1.assigned(taskId), "Task manager 1 should be assigned");
+                    assert.ok(taskManager1.subscribed(taskId), "Task manager 1 should be subscribed");
+                });
+
+                it("Can abandon a subscribed task before attach", async () => {
+                    const taskId = "taskId";
+                    taskManager1.subscribeToTask(taskId);
+                    containerRuntimeFactory.processAllMessages();
+
+                    taskManager1.abandon(taskId);
+                    containerRuntimeFactory.processAllMessages();
+
+                    assert.ok(!taskManager1.queued(taskId), "Task manager 1 should not be queued");
+                    assert.ok(!taskManager1.assigned(taskId), "Task manager 1 should not be assigned");
+                    assert.ok(!taskManager1.subscribed(taskId), "Task manager 1 should not be subscribed");
+                });
+            });
+
+            describe("Completing tasks", () => {
+                it("Can complete a task before attach", async () => {
+                    const taskId = "taskId";
+                    const volunteerTaskP = taskManager1.volunteerForTask(taskId);
+                    containerRuntimeFactory.processAllMessages();
+                    await volunteerTaskP;
+                    assert.ok(taskManager1.assigned(taskId), "Should be assigned");
+
+                    taskManager1.complete(taskId);
+                    containerRuntimeFactory.processAllMessages();
+                    assert.ok(!taskManager1.queued(taskId), "Should not be queued");
+                    assert.ok(!taskManager1.assigned(taskId), "Should not be assigned");
+                });
+
+                it("Can emit completed event before attach", async () => {
+                    const taskId = "taskId";
+                    const volunteerTaskP1 = taskManager1.volunteerForTask(taskId);
+
+                    containerRuntimeFactory.processAllMessages();
+                    await volunteerTaskP1;
+
+                    let taskManager1EventFired = false;
+                    taskManager1.once("completed", (completedTaskId: string) => {
```

Comment:
Here and elsewhere you'll need to use "on" rather than "once" to test for double-fire